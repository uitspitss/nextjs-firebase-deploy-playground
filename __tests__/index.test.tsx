import React from "react";
import { mount } from "enzyme";
import { render } from "@testing-library/react";

import App from "#/pages/index";

describe("With Enzyme", () => {
  it('Shows "ðŸš€"', () => {
    const app = mount(<App />);
    expect(app.find("p").text()).toEqual("ðŸš€");
  });
});

describe("With React Testing Library", () => {
  it('Shows "ðŸš€"', () => {
    const { getByText } = render(<App />);
    expect(getByText("ðŸš€")).not.toBeNull();
  });
});

describe("With React Testing Library SnapShot", () => {
  it("Should match Snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
