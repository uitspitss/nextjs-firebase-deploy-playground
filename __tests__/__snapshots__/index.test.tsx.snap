// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library SnapShot Should match Snapshot 1`] = `
<DocumentFragment>
  <main>
    <header>
      <a
        class=""
        href="/"
      >
        Home
      </a>
      <a
        class=""
        href="/about"
      >
        About
      </a>
    </header>
    <p
      class="css-1lnatx3"
    >
      ðŸš€
    </p>
  </main>
</DocumentFragment>
`;
